### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by fritz-hh**
Hi Gaurav,

This is the 1st of the 3 panned pull requests.
This pull request does not add any feature, but contains a refactoring to prepare the next 2 pull requests: recursive check, and hyperlinks check.
I will create the next 2 pull requests (based on your master) once this pull request one is merged.

Content:
* Improved notification to mention the name of the file that contained the bad reference (important in case of recursive checks where many adoc files are checked)
* code quality
  * reducing nesting to a minimum to improve readability and maintainability (without changing the overall structure)
    * using Invert “if” statement where possible (see: https://stackoverflow.com/questions/268132/invert-if-statement-to-reduce-nesting)
    * using if, elif, else, rather than nested ifs when possible
    * move some function definition higher
  * consistant impementation of all pattern matching sections
  * make use of the promise resolve method to pass results rather than using a global variable "allGood"
* Various robustness improvements

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
Thank you @fritz-hh 

I am traveling this week so this may take some time. This does not mean I am not looking at this, I will keep working on this but my responses will be delayed. Thanks again, I cannot help but ask you, would you like to be a contributor for this project, since you suggested and made significant contributions? 😃

**Comment by fritz-hh** [comment]
Hi @gaurav-nelson ,

Yes, you can add me as a contributor to the project.
Are there special rules you want me to follow as a collaborator to the project?
Wish you a good travel.

File: lib/asciidoc-reference-check.js [hunk 1]
```diff
@@ -1,456 +1,369 @@
...
+      );
+    } else if (notFoundInternal[0] == null) {
+        atom.notifications.addSuccess(
+          '`' + logMsgFilePath + '`: All internal references are `OK`.'
+        );
+    } else {
+      notFoundInternal.forEach(function (it) {
+        //console.log('Cannot find the anchor: ' + it + ' in current file.');
+        atom.notifications.addError(
+          '`' + logMsgFilePath + '`: Cannot find anchor: `' + it + '` in current file.',
```
Comment (gaurav-nelson) [inline lib/asciidoc-reference-check.js:336]: With the updated notification, which has filename in the message itself, I guess we can change this to
```
'`' + logMsgFilePath + '`: Cannot find anchor: `' + it + '` .',
```

**Comment by gaurav-nelson** [comment]
Thank you @fritz-hh these updates works like charm. I have just made one suggestion which you can tackle in the next PRs. Merging this now. I will wait for the new PRs before I push these onto atom package repo.

**Comment by gaurav-nelson** [comment]
@fritz-hh I have given you Collaborator access for this project, so now you can push things yourself. I do not have any guidelines set up, but the only thing I can think of right now is to get a review from each other before merging. Welcome aboard 👍 🛩 😄 🍺

File: lib/asciidoc-reference-check.js (hunk 1 above)
Comment (fritz-hh) [reply lib/asciidoc-reference-check.js:336]: I agree with you. It will be changed in the next pull request

**Pull request merged**